frame/staking/src/lib.rs

```diff
@@ -732,6 +731,143 @@ impl<AccountId, Balance: Default + HasCompact> Default for Exposure<AccountId, B
 	}
 }
 
+impl<AccountId: Clone, Balance: HasCompact + AtLeast32BitUnsigned + Copy>
+	Exposure<AccountId, Balance>
+{
+	/// Splits an `Exposure` into `ExposureOverview` and multiple chunks of `IndividualExposure`
+	/// with each chunk having maximum of `page_size` elements.
+	fn into_pages(
+		self,
+		page_size: u32,
+	) -> (ExposureOverview<Balance>, Vec<ExposurePage<AccountId, Balance>>) {
+		let individual_chunks = self.others.chunks(page_size as usize);
+		let mut exposure_pages: Vec<ExposurePage<AccountId, Balance>> =
+			Vec::with_capacity(individual_chunks.len());
+
+		for chunk in individual_chunks {
+			let mut page_total: Balance = Zero::zero();
+			for individual in chunk.iter() {
+				page_total = page_total.saturating_add(individual.value);
+			}
+
+			exposure_pages.push(ExposurePage {
+				page_total,
+				others: chunk
+					.iter()
+					.map(|c| IndividualExposure { who: c.who.clone(), value: c.value })
+					.collect(),
+			});
+		}
+
+		(
+			ExposureOverview {
+				total: self.total,
+				own: self.own,
+				nominator_count: self.others.len() as u32,
+				page_count: exposure_pages.len() as PageIndex,
+			},
+			exposure_pages,
+		)
+	}
+}
+
+/// A snapshot of the stake backing a single validator in the system.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct ExposurePage<AccountId, Balance: HasCompact> {
+	/// The total balance of this chunk/page.
+	#[codec(compact)]
+	pub page_total: Balance,
+	/// The portions of nominators stashes that are exposed.
+	/// TODO(ank4n): BoundedVec touches lot of code, skip for now.
```

Comment:
Make sure to add a follow-up issue and add the issue link here (same for any leftover `TODO` in the code).